### DISCUSSION TITLE
'DYNAMIC_SERVER_USAGE' error - generateStaticParams

### DISCUSSION THREAD

**Discussion by WillEtheridge** [vercel/next.js#64438] [Help]
### Summary

Hi all.

I'm building a site that displays meal plans to users.

I'm trying to use generateStaticParams to create the routes to the pages.

Things are working fine when I run the app locally however when I build the app I receive the following error when trying to access one of the dynamic routes.

 ⨯ [Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: 'DYNAMIC_SERVER_USAGE'
}

Here is a simplified version of my code that I am getting the error with.

```javascript
export async function generateStaticParams() {
  const caloriesOptions = [1200, 1500, 1800]; // Example calorie values
  return caloriesOptions.map(calorie => ({
    params: { calories: String(calorie) },
  }));
}

export default function CaloriesPage({ params }) {
  return (
    <div>
      <h1>{params.calories} Calorie Meal Plan</h1>
      <p>Details about the {params.calories} calorie meal plan go here.</p>
    </div>
  );
}

``` 

There's something that I'm missing or not getting here but I can't for the life of me figure out what. Any help or guidance in the right direction is much appreciated.

Thanks!

### Additional information

```js
Next config

/** @type {import('next').NextConfig} */
const nextConfig = {};

export default nextConfig;


package.json
{
  "name": "ai-meal-planner",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@anthropic-ai/sdk": "^0.17.1",
    "@clerk/nextjs": "^4.29.9",
    "@next/third-parties": "^14.1.4",
    "@stripe/stripe-js": "^3.2.0",
    "html2canvas": "^1.4.1",
    "jspdf": "^2.5.1",
    "mongodb": "^6.5.0",
    "next": "^14.1.3",
    "openai": "^4.29.1",
    "puppeteer": "^22.6.1",
    "react": "^18",
    "react-dom": "^18",
    "react-icons": "^5.0.1",
    "stripe": "^14.23.0",
    "svix": "^1.21.0"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10.0.1",
    "postcss": "^8",
    "tailwindcss": "^3.3.0",
    "typescript": "^5"
  }
}
```

**Comment by mkmyhre** [comment]
Same issue, did you solve this?

**Comment by kevinkam** [comment]
I faced about same issue, but only happen if the generateStaticParams is returning empty array
so for empty array case I put `dynamic = "force-dynamic"`, and the other case I put `dynamic = "auto"`  then it fixed

**Reply by kevinkam** [reply]
IDE will give error if the `export const dynamic` is conditional, but when build it wont throw error

**Comment by channyeintun** [comment]
I'm also experiencing this.
